oaknational/Oak-Web-Application
# Feat/authenticated requests

## Description

- Separates graphql clients into `node-lib/graphql` and `browser-lib/graphql`
- deletes "firebase" cloud functions folder, since now that logic is done in a next api route ("/api/confirm-new-user")
- adds `node-lib/auth.ts` which connects to the `firebase-admin` api
- adds `pages/api/auth/confirm-new-user.ts`
- adds `LoadingSpinner` component
- updates `config/index.ts` to improve functionality, differentiating between `NEXT_PUBLIC_` and other vars
- adds `BookmarksProvider` and `useBookmarks` 
- adds `pages/bookmarks.tsx`

## Issue(s)

Fixes #80

## How to test

1. Go to https://samara-lz901qfhc-oak-national-academy.vercel.app/
2. Click "Sign in" in the top riht and go through the flow
3.Open [a lesson](https://samara-lz901qfhc-oak-national-academy.vercel.app/lessons/physics-only-review-chj3cd) or [another lesson](https://samara-lz901qfhc-oak-national-academy.vercel.app/lessons/lady-macbeth-c9jk2c) 
4. Click on "Add bookmark"
5. Go to the [bookmarks page](https://samara-lz901qfhc-oak-national-academy.vercel.app/bookmarks)
6. You should see your bookmarks
7. Click the "delete" button next to the bookmark
8. It should disappear

**NB still some work to do on error handling, and the case where the user is not logged in, etc**

## Screenshots


How it should now look:
{screenshots}
